### DISCUSSION TITLE
这个堆栈信息是不是去掉更友好？

### DISCUSSION THREAD

**Discussion by chenlinzhong** [apache/doris#9098] [RFC]
堆栈目的是助力查问题，线上一般的做法是

- 1.开启core
- 2.如果出于性能的考虑，不开启core，一般是在程序退出的时候打印core栈

现在报错都默认打印了栈信息，太密集了，占满了整个屏幕，不利于看日志排查问题

![image](https://user-images.githubusercontent.com/11487604/163940090-bba7777a-a2bc-40bb-a130-949c578786fe.png)
![image](https://user-images.githubusercontent.com/11487604/163940186-38aed8f2-0f00-4e85-9275-9eae24efe60d.png)
![image](https://user-images.githubusercontent.com/11487604/163940296-29c32aef-fa64-4558-923f-bf7eb52375fd.png)

**Comment by yiguolei** [comment]
Stack is used not only for core, but also to detect the potential bugs or errors in the system. For example, if user meets an error during query and BE print error messages, we could not find the call stack if we do not read the code and the code did not print log for every call line.  You could read this discussion https://github.com/apache/incubator-doris/discussions/8406 I have explained why do we need using exception stack intread of just an error code.

Currently, doris print many logs to get the exception stack, it could be deleted because use have exception stack in status.

For your case, I think there is an improvement in Status.h. M(OLAP_ERR_CUMULATIVE_NO_SUITABLE_VERSION, -2000, "", true) \ maybe we should set this errorcode not print exception stack. You could set it to M(OLAP_ERR_CUMULATIVE_NO_SUITABLE_VERSION, -2000, "", false).

@chenlinzhong This is a very good PR, could you please submit a PR to Doris?

**Reply by chenlinzhong** [reply]
it's a good job to unified status code，but for stack I have these questions：

- The above status is just an example. It may also be that other statuses
- At present, all States expect "OK" will print the stack by default. The stack content is too long, and the size of almost every stack exceeds 2K. Writing to the disk in high concurrency scenarios may damage the performance
- The log is not standardized, A log is printed across lines，this is a unusual  way
- As for troubleshooting, from past experience, it is basically enough to have error reporting information. Moreover, this stack has only call information and no context. It doesn't feel that it is very helpful. You can understand the error reporting. Usually, you are familiar with this code，If you are not familiar with this code, stack is meaningless

**one log size**
![7a4bb8fb6e90b5932a2eb76f3e27c4c9](https://user-images.githubusercontent.com/11487604/164428280-fd19c6fb-66a4-4f36-ac23-14c974cc2ece.png)

**too long, trigger printing a warn log** 
![6ffebf3a4b5d9a1ccb7769abcb29d1d1](https://user-images.githubusercontent.com/11487604/164427563-febb728f-4f84-4816-a967-c51b290baefc.png)

**Comment by chenlinzhong** [comment]
how about  add switch control? 

https://github.com/apache/incubator-doris/pull/9132/files

@yiguolei

**Reply by yiguolei** [reply]
1. Error log should not be printed too many times just like throw exception in java. If the error status is constructed, it should be error that should attention.
2. Currently, there are many log mechanism in Doris, such as LOG(WARNING) and OLAP_LOG_WARNING, we should also unify them. And the log format is not very well, should also be improved.
3. For your question 4, I am no agree, currently doris print so many logs to get the error stack, many logs could be deleted. For example, if method A call method B, method B call method C, method C call method D. If D meets error, you will find that there are log messages in A,B,C to print the error. If we have error stack info in Status, log in method B,C could be deleted, only log in A is kept.

**Reply by yiguolei** [reply]
@chenlinzhong Do you have wechat? we could discuss it.

**Reply by chenlinzhong** [reply]
ok, lylwa928

**Reply by chenlinzhong** [reply]
@yiguolei 

<img width="1294" alt="image" src="https://user-images.githubusercontent.com/11487604/165085987-5fdb57aa-79f8-40ba-8e43-6c646459ce3d.png">

**Comment by LOVEGISER** [comment]
1.i think stack log is very important for detect doris runtime status.
2.if stack log is too large,suggest split and  clear log timer by config.
